momepy/functional/_distribution.py

```diff
@@ -125,10 +133,19 @@ def shared_walls(geometry: GeoDataFrame | GeoSeries) -> Series:
     143    10.876113
     Name: shared_walls, Length: 144, dtype: float64
     """
+
+    predicate = "touches"
+    if not strict:
+        orig_lengths = geometry.length
+        geometry = geometry.buffer(tolerance)
+        predicate = "intersects"
+
     if GPD_GE_013:
-        inp, res = geometry.sindex.query(geometry.geometry, predicate="touches")
+        inp, res = geometry.sindex.query(geometry.geometry, predicate=predicate)
     else:
-        inp, res = geometry.sindex.query_bulk(geometry.geometry, predicate="touches")
+        inp, res = geometry.sindex.query_bulk(geometry.geometry, predicate=predicate)
+
+    inp, res = inp[inp != res], res[inp != res]
```

Comment:
```suggestion
    mask =  inp != res
    inp, res = inp[mask], res[mask]
```
minor difference but avoids one computation